chainer/functions/array/hstack.py

```diff
@@ -53,11 +53,40 @@ def hstack(xs):
     """Concatenate variables horizontally (column wise).
 
     Args:
-        xs (list of chainer.Variable): Variables to be concatenated.
+        xs (list of :class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`):
+            Input variables to be concatenated. The variables must have the
+            same ``ndim``. Also, when the variables have horizontal dimension
+            (i.e. :math:`ndim \\geq 2`), the variables must have the same
+            shape, except in the horizontal (column wise) dimension.
```

Comment:
I am afraid this is not the right behavior, unfortunately. When `ndim=1`, this function concatenates along the first axis and the input variables need not have same shape.

```
In [2]: a = np.array([3, 4])
In [3]: b = np.array([5, 6, 7, 8])
In [6]: chainer.functions.hstack((a, b)).data
Out[6]: array([3, 4, 5, 6, 7, 8])
```